src/java.base/share/classes/java/lang/Process.java

```diff
@@ -149,12 +169,107 @@ public abstract class Process {
     private Charset inputCharset;
     private BufferedReader errorReader;
     private Charset errorCharset;
+    private boolean closed;     // true if close() has been called
 
     /**
      * Default constructor for Process.
      */
     public Process() {}
 
+    /**
+     * Close all writer and reader streams immediately and wait for the process to terminate.
+     * This method is idempotent, if this {@code Process} has already been closed
+     * invoking this method has no effect.
+     * <p>
+     * If the data from the process input or error streams is needed, it must be read before
+     * calling this method. The contents of streams that have not been read to end of stream
+     * are lost, they are discarded or ignored.
+     * <p>
+     * If the process exit value of the process is of interest, then the caller must
+     * {@linkplain #waitFor() wait} for the process to terminate before calling this method.
+     * <p>
+     * Streams should be closed when no longer needed.
+     * Closing an already closed stream usually has no effect but is specific to the stream.
+     * If an {@code IOException} occurs when closing a stream it is thrown
+     * when this method returns. Additional {@code IOExceptions}
```

Comment:
What would you think of replacing "when this method returns" with "after the process has terminated"?